### ISSUE TITLE
No fallback in circuit_drawer

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#1077]
The `circuit_drawer` falls back to `_matplotlib_circuit_drawer` when `_latex_circuit_drawer` fails with `OSError`. We should remove that fallback at some point.

**Comment by mtreinish** [comment]
Yep, I agree and that's the plan. Which is why we emit a deprecation warning right now if you run `circuit_drawer` without the `output` kwarg set. This was the deprecated previous default for `circuit_drawer` in 0.6.0. We will have to wait until we at least have this emit the deprecation warning in at least 1 release.

It does raise a larger question for me though which is what our deprecation policy should be on this kind of thing? I haven't seen it defined anywhere where we say that we wait X amount of time (whether in an actual unit of time, or a release count) from deprecation to removal. I think that's a separate larger discussion though, but something I think that will be good to define as a policy for the project.

**Comment by jaygambetta** [comment]
I would like the output of the cirucit_drawer to be text (no dependencies by default). 

We can have a visitation.set_defaults for interactive when we have it etc

**Comment by 1ucian0** [comment]
Maybe `visitation.set_defaults` can also be used for handle the drawer options (see #1075). Opinions?

**Comment by jaygambetta** [comment]
Did we decide on this @1ucian0 and @mtreinish I would like to make sure that we are moving away from fallbacks? If this is done let's close if it is getting worked on who is it assigned to.

**Comment by mtreinish** [comment]
This is basically done, we've deprecated the fallback as the default and as soon as 0.7 is out I'm going to push a patch up removing the fallback and switching to 'text' output by default for 0.8.

**Comment by 1ucian0** [comment]
We can say that it's decided :)
